### PR TITLE
fix: 🐛 icons multi provider

### PR DESCRIPTION
**Pull request by kirills98**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

If has two modules (A and B) that use `SvgIconsModule.forChild()` and this modules imports to module C then will be provided only icons from last imported module (for example B) and icons from other modules ( for example A) will be ignored.  

Issue Number: N/A

## What is the new behavior?

Now icons provider with `multi: true` parameter. It's mean that all provided icons will accessible as array by same token.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by NetanelBasal** [comment]
I'm not sure what you're trying to solve. Please add specs.

**Comment by kirills98** [comment]
> I'm not sure what you're trying to solve. Please add specs.

I make [small demo](https://stackblitz.com/edit/angular-lib-token-providing?file=src/app/lib/lib.module.ts) for this case.  

My real case: i have several reusable components that uses inside `SvgIconsModule.forChild()`. If this components used together, then icon registering incorectly.

**Comment by NetanelBasal** [comment]
[forChild](https://github.com/ngneat/svg-icon#lazy-loaded-modules) is meant to be used only in lazy load modules. 

In your example, they are eager. Do you want to use it in eager modules just to make it more organized?

**Comment by kirills98** [comment]
@NetanelBasal yes. Because one of case using icons in external angular package (bundled by ng-packagr).  
Of course we can make some our module for this or we can register icons manually. But i think it will good "out of box" :)

File: projects/ngneat/svg-icon/src/lib/svg-icons.module.ts [hunk 1]
```diff
@@ -24,17 +24,17 @@ export class SvgIconsModule {
   static forChild(icons: SvgIconType | SvgIconType[]): ModuleWithProviders<SvgIconsModule> {
     return {
       ngModule: SvgIconsModule,
-      providers: [{ provide: SVG_ICONS, useValue: icons }],
+      providers: [{ provide: SVG_ICONS, useValue: icons, multi: true }],
     };
   }
 
   constructor(
-    @Optional() @Inject(SVG_ICONS) icons: SvgIconType | SvgIconType[],
```
Comment (NetanelBasal) [inline projects/ngneat/svg-icon/src/lib/svg-icons.module.ts:32]: This is a breaking change

**Review by NetanelBasal — APPROVED** [review]

**Comment by NetanelBasal** [comment]
Please update the docs

**Comment by kirills98** [comment]
@NetanelBasal i'm fixed docs, additionaly add `@Self` decorator for injecting icons in module, for improve re-register icons, provided in parent modules

**Pull request merged**